plugins/partiql-local/src/main/kotlin/org/partiql/plugins/local/LocalCatalog.kt

```diff
@@ -1,109 +1,109 @@
-/*
- * Copyright Amazon.com, Inc. or its affiliates.  All rights reserved.
- *
- *  Licensed under the Apache License, Version 2.0 (the "License").
- *  You may not use this file except in compliance with the License.
- *  A copy of the License is located at:
- *
- *       http://aws.amazon.com/apache2.0/
- *
- *  or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
- *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
- *  language governing permissions and limitations under the License.
- */
-
 package org.partiql.plugins.local
 
-import com.amazon.ionelement.api.loadSingleElement
-import org.partiql.spi.BindingCase
-import org.partiql.spi.BindingName
-import org.partiql.spi.BindingPath
-import org.partiql.types.StaticType
-import java.io.File
+import org.partiql.planner.catalog.Catalog
+import org.partiql.planner.catalog.Function
+import org.partiql.planner.catalog.Identifier
+import org.partiql.planner.catalog.Name
+import org.partiql.planner.catalog.Namespace
+import org.partiql.planner.catalog.Session
+import org.partiql.planner.catalog.Table
 import java.nio.file.Path
+import kotlin.io.path.isDirectory
+import kotlin.io.path.nameWithoutExtension
+import kotlin.io.path.notExists
 
-private sealed class FsTree(val name: String) {
+/**
+ * Implementation of [Catalog] where dirs are namespaces and files are table metadata.
+ */
+internal class LocalCatalog(
+    private val name: String,
+    private val root: Path,
+) : Catalog {
 
-    // "Directory" node
-    class D(name: String, val children: List<FsTree>) : FsTree(name)
+    private companion object {
+        private const val EXT = ".ion"
+    }
 
-    // Type node
-    class T(name: String, val type: StaticType) : FsTree(name)
-}
+    init {
+        assert(root.isDirectory()) { "LocalNamespace must be a directory" }
+    }
 
-/**
- * Build a memoized catalog tree from local schema definitions.
- */
-internal class LocalCatalog private constructor(private val root: FsTree.D) {
+    override fun getName(): String {
+        return name
+    }
+
+    override fun getTable(session: Session, name: Name): Table? {
+        val path = toPath(name.getNamespace()).resolve(name.getName() + EXT)
+        if (path.notExists() || !path.isDirectory()) {
+            return null
+        }
```

Comment:
```suggestion
        if (path.notExists() || path.isDirectory()) {
            return null
        }
```
If I'm reading this correctly.